database/migrations/20220531100155-update-sites.js

```diff
@@ -0,0 +1,195 @@
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    /**
+     * Add altering commands here.
+     *
+     * Example:
+     * await queryInterface.createTable('users', { id: Sequelize.INTEGER });
+     */
+    const transaction = await queryInterface.sequelize.transaction()
+    try {
+      await queryInterface.dropTable("site_members", { transaction })
+      await queryInterface.dropTable("sites", { transaction })
+
+      await queryInterface.createTable(
+        "sites",
+        {
+          id: {
+            allowNull: false,
+            autoIncrement: true,
+            primaryKey: true,
+            type: Sequelize.BIGINT,
+          },
+          repository_name: {
+            allowNull: false,
+            unique: true,
+            type: Sequelize.TEXT,
+          },
+          created_by: Sequelize.TEXT,
```

Comment:
I agree that this would be ideal, but it will be a bit dangerous to put in this constraint today. Form SG can't check that the form submitter is in the database. If the submitter isn't in the database, then the database insert will fail after the rest of the site creation has succeeded. The only way to recover would be to extract the site information from the error email and insert it into the database manually. Is this good enough? If not, then we should probably wait to do something like this until the day when we can put this interface in the cms frontend.

Reply:
this data model **doesn't** have to be enforced on the database layer - we can have a service method that looks like so: 

```ts
const addCreatedBy = (user: UnverifiedUser | VerifiedUser) => {...}
```

where the type is either 

```ts
interface User {
    contact_number?: number
    email?: Email // If we are adapting this, we need to validate email @ routes
    name: string 
}

interface UnverifiedUser extends User { 
    type: "unverified" 
}

interface VerifiedUser extends User { 
    type: "verified" 
    // additional properties
} 
```

but i have to ask - if the email is in the database, we could use a FK to link to an existing user then add a form submitter property to reference this or just check that the user tagged to the existing email is the same as that of the form submitter (and make it `VerifiedUser` else `UnverifiedUser`)